File: flutter_reactive_ble/ios/Classes/BleData extras/BLEStatus.swift
import enum CoreBluetooth.CBManagerState

func encode(_ centralState: CBManagerState) -> Int32 {
    switch (centralState) {
    case .unknown, .resetting:
        return 0
    case .unsupported:
        return 1
    case .unauthorized:
        return 2
    case .poweredOff:
        return 3
    case .poweredOn:
        return 5
    }
}

File: packages/reactive_ble_platform_interface/lib/src/model/ble_status.dart
/// The status of the BLE subsystem as reported by the platform API.
enum BleStatus {
  /// Status is not (yet) determined.
  unknown,

  /// BLE is not supported on this device.
  unsupported,

  /// BLE usage is not authorized for this app.
  unauthorized,

  /// BLE is turned off.
  poweredOff,

  /// Android only: Location services are disabled.
  locationServicesDisabled,

  /// BLE is fully operating for this app.
  ready
}
